server/verifiers/twitter-verifier/verifier/src/main/java/org/kaaproject/kaa/server/verifiers/twitter/verifier/TwitterUserVerifier.java

```diff
@@ -0,0 +1,208 @@
+                        LOG.debug("Connection error: can't close CloseableHttpResponse");
+                    }
+                }
+            }
+        }
+    }
+
+    private void handleResponse(CloseableHttpResponse connection, String userExternalId, UserVerifierCallback callback,
+                                String userAccessToken) throws IOException {
+        Map<String, Object> responseMap = getResponseMap(connection.getEntity().getContent());
+        String receivedUserId = String.valueOf(responseMap.get(ID));
+
+        if (!receivedUserId.equals(userExternalId)) {
+            LOG.warn("Input token doesn't belong to the user with {} id", userExternalId);
+            callback.onVerificationFailure("User access token " + userAccessToken + " doesn't belong to the user");
+        } else {
+            LOG.trace("Input token is confirmed and belongs to the user with {} id", userExternalId);
+            callback.onSuccess();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void handleBadResponse(CloseableHttpResponse connection, UserVerifierCallback callback) throws IOException {
+        Map<String, Object> responseMap = getResponseMap(connection.getEntity().getContent());
+        List<Map<String, Object>> errorList = (List<Map<String, Object>>) responseMap.get(ERRORS);
+
+        // there is only one error in the list
+        Map<String, Object> error = errorList.get(0);
+
+        if (INVALID_TOKEN_CODE.equals(String.valueOf(error.get(CODE)))) {
+            LOG.trace("Access Token is invalid or expired");
+            callback.onTokenInvalid();
+        } else {
+            LOG.trace("Unable to verify token. Error code: [{}], message[{}]", error.get(CODE), error.get(MESSAGE));
+            callback.onVerificationFailure("Unable to verify token. Error code: " + error.get(CODE) +
+                    ", message: " + error.get(MESSAGE));
+        }
+    }
+
+    private Map<String, Object> getResponseMap(InputStream inputStream) throws IOException {
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+            ObjectMapper responseMapper = new ObjectMapper();
+            return responseMapper.readValue(reader.readLine(), Map.class);
+        }
+    }
+
+    protected CloseableHttpResponse establishConnection(String tokenAndSecret) throws IOException {
+        HttpRequest request = null;
+        try {
+            request = new BasicHttpEntityEnclosingRequest(REQUEST_METHOD, TWITTER_PATH);
+
+            String[] tokenThenSecret = tokenAndSecret.split(" ");   // now user access token is tokenThenSecret[0]
+                                                                    // and user secret is tokenAndSecret[1]
+
+            request.setHeader(REQUEST_HEADER_NAME,
+                    oAuthHeaderBuilder.generateHeader(tokenThenSecret[0], tokenThenSecret[1]));
+        } catch (InvalidKeyException e) {
+            LOG.debug("Invalid key for Encryption", e);
```

Comment:
Never do this. Propagate this exception and report is as internal appender error